src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/AccessTokenValidator.cs

```diff
@@ -0,0 +1,75 @@
+﻿using AirBnB.Domain.Entities;
+using AirBnB.Domain.Enums;
+using AirBnB.Infrastructure.Common.Settings;
+using FluentValidation;
+using Microsoft.Extensions.Options;
+
+namespace AirBnB.Infrastructure.Common.Validators;
+
+/// <summary>
+/// Validator for the AccessToken entity using FluentValidation.
+/// </summary>
+public class AccessTokenValidator : AbstractValidator<AccessToken>
+{
+    /// <summary>
+    /// Initializes a new instance of the AccessTokenValidator class.
+    /// </summary>
+    /// <param name="jwtSettings">Options for configuring JWT settings injected via dependency injection.</param>
+    public AccessTokenValidator(IOptions<JwtSettings> jwtSettings)
+    {
+        var jwtSettingsValue = jwtSettings.Value;
+
+        // Validation rules for the 'OnCreate' scenario.
+        RuleSet(
+            EntityEvent.OnCreate.ToString(),
+            () =>
+            {
+                // Ensure that IsRevoked is not set to true during creation.
+                RuleFor(accessToken => accessToken.IsRevoked).NotEqual(true);
+
+                // Ensure that UserId is not empty during creation.
+                RuleFor(accessToken => accessToken.UserId).NotEqual(Guid.Empty);
+            }
```

Comment:
Shouldn't we check `Token` and `ExpiryTime` properties too ?

Reply:
It has been changed